packages/paste-core/components/meter/src/index.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import {type BoxProps, Box} from '@twilio-paste/box';
+import {Label} from '@twilio-paste/label';
+import {Text} from '@twilio-paste/text';
+import type {HTMLPasteProps} from '@twilio-paste/types';
+import {useMeter} from '@twilio-paste/react-spectrum-library';
+
+export interface MeterProps extends HTMLPasteProps<'meter'>, Pick<BoxProps, 'element'> {
+  minValue?: number;
+  maxValue?: number;
+  value?: number;
+  label: string;
+  showValueLabel?: boolean;
+  formatOptions?: Intl.NumberFormatOptions; // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options
+  valueLabel?: string;
+  /*
+   * The following props don't exist on the react-aria useMeter hook but do exist on the HTML meter element.
+   * They can be added back into the Paste Meter API depending on the spec & designs.
+   *
+   * low?: number;
+   * high?: number;
+   * optimum?: number;
+   */
+}
+
+const Meter = React.forwardRef<HTMLMeterElement, MeterProps>(({element = 'METER', ...props}, ref) => {
+  const {label, showValueLabel = Boolean(label), value = 0, minValue = 0, maxValue = 100} = props;
+  const {meterProps, labelProps} = useMeter(props);
+
+  // Calculate the width of the bar as a percentage
+  const percentage = (value - minValue) / (maxValue - minValue);
+  const barWidth = `${Math.round(percentage * 100)}%`;
+
+  return (
+    <Box as="div" {...meterProps} role="meter" maxWidth="size30" element={`${element}_WRAPPER`}>
```

Comment:
thought: I wonder if this element should be `METER` and then the bar is `METER_BAR` and `METER_FILL`? Especially if the Label stuff is pulled out into a composition

Reply:
Yeah, this is the main thing I was hoping for feedback on. Even though this element is technically the meter, it includes both labels so I named it `METER_WRAPPER` because I figured the naming for customization was purely for applying styles and would make more sense to consumers who are trying to style the whole component, including the labels, even though it's not _technically_ the meter itself. But that being said I like your suggestion too and definitely can go in that direction!

Reply:
FWIW if you look at the shadow dom of the native element, it's not too disimilar 

<img width="580" alt="image" src="https://github.com/twilio-labs/paste/assets/368249/63076041-56f4-4218-97b8-d8e15edca166">

The outer element is the `meter` and then there are sub parts including the bar.